src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,200 @@
+        return cart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE cart_id = ? AND deleted = FALSE";
+        ShoppingCart cart = null;
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                cart = getCartFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get shopping cart with id = "
+                    + id, ex);
+        }
+
+        if (cart != null) {
+            cart.setProducts(getListOfProducts(cart.getId()));
+        }
+        return Optional.ofNullable(cart);
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT * FROM shopping_carts WHERE deleted = FALSE";
+        List<ShoppingCart> carts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart cart = getCartFromResultSet(resultSet);
+                carts.add(cart);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get list of all shopping carts", ex);
+        }
+
+        for (ShoppingCart cart: carts) {
+            cart.setProducts(getListOfProducts(cart.getId()));
+        }
+        return carts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        String query = "UPDATE shopping_carts SET user_id = ? "
+                + "WHERE cart_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cart.getUserId());
+            statement.setLong(2, cart.getId());
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update shopping cart with id = "
+                    + cart.getId(), ex);
+        }
+
+        removeAllProductsFromShoppingCart(cart.getId());
+        insertProductsToShoppingCart(cart);
+        return cart;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE shopping_carts SET deleted = TRUE WHERE cart_id = ?";
+        int itemsDeleted = 0;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            itemsDeleted = statement.executeUpdate();
```

Comment:
the same as somewhere above - return statement.execute.Update == 1; and remove variable itemsDeleted...